pages/generator.php

```diff
@@ -11,22 +11,43 @@
  * - Page will return blank if stuff works out as expected.
  */
 
-// generate page cache
-$pages = rex_get('pages', 'string');
-if (!empty($pages)) {
-    $generator = new cache_warmup_generator_pages;
-    $items = cache_warmup_generator::prepareItems($pages);
-    $generator->generateCache($items);
+$proceed = true;
+
+if (class_exists('rex_csrf_token')) {
+
+    $mode = rex::getProperty('redaxo');
+    rex::setProperty('redaxo', true); // switch REDAXO to backend mode
+    if (!rex_csrf_token::factory('cache_warmup_generator')->isValid()) {
+        $proceed = false; // token not valid, do not proceed!
+    }
+    rex::setProperty('redaxo', $mode); // reset to previous mode
```

Comment:
solche temporären um + zurückschalter würde ich mit einem try/finally machen

Reply:
Okay, verstehe. Bevor ich mich in dem PR verfranse, möchte ich lieber erstmal mergen und später gemeinsam Quality machen bzw durch PRs von Leuten wie dir lernen, wie man sowas sauber umsetzt.